## Задача 1-1. 
Из колоды в 52 карты извлекаются случайным образом 4 карты.

a) Найти вероятность того, что все карты – крести.

### Решение:
[1] Число возможных комбинаций 4 карты крести:
$$C_{13}^4=\frac{13!}{4!(13-4)!}=\frac{9!*10*11*12*13}{4!*9!}$$

In [20]:
from math import factorial as fact
def comb(n, k):
    return int(fact(n) / (fact(k) * fact(n - k)))
c_4_13 = comb(13, 4)
print(c_4_13)

715


[2] Общее число возможных комбинаций 4 карт:
$$C_{52}^4=\frac{52!}{4!(52-4)!}=\frac{48!*49*50*51*52}{4!48!}$$

In [12]:
c_4_52 = comb(52, 4)
print(c_4_52)

270725


Вероятность вытащить 4 крестовых карты $$P(A)=\frac{число-благоприятных-исходов}{общее-число-исходов}=\frac{715}{270725}=0,002641 $$

In [17]:
r = float(c_4_13 / c_4_52)
print(r*100)

0.26410564225690275


#### Ответ 1-1а: вероятность вытащить 4 крестовых карты составляет 0,2641%

б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

### Решение:
[1] Найдем все сочетания 4-х карт без тузов (52 карты - 4 туза = 48) $C_{48}^4$:

In [11]:
c_4_48 = comb(48, 4)
print(c_4_48)

194580


[2] Число исходов в котрых хотя бы 1 туз (или 2, или 3, или 4) $$C_T =C_{52}-C_{48}^4=270725-194580=76145$$

In [18]:
c_1_4 = c_4_52 - c_4_48
print(c_1_4)

76145


$P(A)= \frac{76145}{270725}$

In [19]:
prob_min_one_ace = c_1_4 / c_4_52
print(prob_min_one_ace * 100)

28.126327454058547


#### Ответ 1-1б: вероятность, что среди 4-х карт окажется хотя бы один туз составляет 28.126%

## Задача 1-2.
На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

### Решение:
[1] Общее число возможных комбинаций кодового замка:
$$C_{10}^3=\frac{10!}{3!(10-3)!}$$

In [13]:
c_3_10 = comb(10, 3)
print(c_3_10)

120


[2] Так как по условию все три кнопки нужно нажать одновременно => имеем 1 благоприятный исход: $P(A)= \frac{1}{120}$

In [15]:
prob_code = 1 / c_3_10
print(prob_code)

0.008333333333333333


#### Ответ 1-2: вероятность открытия замка с 1 попытки составляет 0.008(3)

## Задача 1-3.
В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

### Решение:
[1] Число сочетаний окрашенных деталей:
$$C_{9}^3=\frac{9!}{3!(9-3)!}$$

In [16]:
c_3_9 = comb(9, 3)
print(c_3_9)

84


[2] Общее число сочетаний деталей в ящике:
$$C_{15}^3=\frac{15!}{3!(15-3)!}$$

In [17]:
c_3_15 = comb(15, 3)
print(c_3_15)

455


[3] Вероятность благоприятного исхода: $P= \frac{C_{9}^3}{C_{15}^3}$

In [18]:
prob_all_painted_balls = c_3_9 / c_3_15
print(prob_all_painted_balls)

0.18461538461538463


##### Проверка:
$$P= \frac{C_{9}^3}{C_{15}^3}=\frac{\frac{9!}{3!(9-3)!}}{\frac{15!}{3!(15-3)!}}=\frac{{9!}*{12!}}{{6!}*{15!}}$$

In [22]:
prob_1_3_test = (fact(9) * fact(12)) / (fact(6) * fact(15))
print(prob_1_3_test)

0.18461538461538463


#### Ответ 1-3: вероятность того, что все извлеченные детали окрашены составляет 0.1846

## Задача 1-4.
В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

### Решение:
[1] Общее число возможных сочетаний выигрышных билетов:
$$C_{100}^2=\frac{100!}{2!(100-2)!}$$

In [23]:
c_2_100 = comb(100, 2)
print(c_2_100)

4950


[2] Вероятность благоприятного исхода: $P= \frac1{C_{100}^2}$

In [25]:
prob_win = 1 / c_2_100
print((prob_win)*100)

0.0202020202020202


#### Ответ 1-4: вероятность того, что 2 приобретенных билета окажутся выигрышными составляет 2,02%